lightning/src/ln/channelmanager.rs

```diff
@@ -3021,6 +3030,24 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		(retain_channel, NotifyOption::DoPersist, ret_err)
 	}
 
+	/// Close the channel if the feerate is non-satisfying to ensure efficient confirmation
+	fn check_channel_autoclose(&self, short_to_id: &mut HashMap<u64, [u8; 32]>, chan_id: &[u8; 32], chan: &mut Channel<Signer>, new_feerate: u32) -> (bool, NotifyOption, Result<(), MsgHandleErrInternal>) {
+		let mut retain_channel = true;
+		let mut notify_option = NotifyOption::SkipPersist;
+		let res = match chan.check_autoclose(new_feerate) {
+			Ok(res) => Ok(res),
+			Err(e) => {
```

Comment:
Honestly this is a lot of error handling code, can we just inline this in timer_tick_occurred? It'd be a lot less total code.